pkg/apply/prune/grouping_test.go

```diff
@@ -581,6 +581,65 @@ func TestAddSuffixToName(t *testing.T) {
 	}
 }
 
+func TestClearGroupingObject(t *testing.T) {
+	tests := map[string]struct {
+		infos   []*resource.Info
+		isError bool
+	}{
+		"Empty infos should error": {
+			infos:   []*resource.Info{},
+			isError: true,
+		},
+		"Non-Unstructured grouping object should error": {
+			infos:   []*resource.Info{nonUnstructuredGroupingInfo},
+			isError: true,
+		},
+		"Info with nil Object should error": {
+			infos:   []*resource.Info{nilInfo},
+			isError: true,
+		},
+		"Single grouping object should work": {
+			infos:   []*resource.Info{copyGroupingInfo()},
+			isError: false,
+		},
+		"Single non-grouping object should error": {
+			infos:   []*resource.Info{pod1Info},
+			isError: true,
+		},
+		"Multiple non-grouping objects should error": {
+			infos:   []*resource.Info{pod1Info, pod2Info},
+			isError: true,
+		},
+		"Grouping object with single inventory object should work": {
+			infos:   []*resource.Info{copyGroupingInfo(), pod1Info},
+			isError: false,
+		},
+	}
+
+	for name, test := range tests {
```

Comment:
Rename the `test` variable to `tc` to avoid the linting errors. The linter doesn't like this pattern because it can lead to bugs, but it is safe in tests like this. I have added an exception in the linter config to ignore this if the variable is named `tc` (which is a common name with this pattern).